kornia/geometry/transform/imgwarp.py

```diff
@@ -102,7 +113,15 @@ def warp_perspective(
     )
     grid = transform_points(src_norm_trans_dst_norm[:, None, None], grid)
 
-    return F.grid_sample(src, grid, align_corners=align_corners, mode=mode, padding_mode=padding_mode)
+    if padding_mode == "fill":
+        # warp a mask of ones, then multiple with padding_tensor and add to src warp
+        ones_mask = torch.ones_like(src)
+        padding_tensor = padding_tensor.to(ones_mask.device)
```

Comment:
```suggestion
        padding_tensor = padding_tensor.to(ones_mask)
```
both `device` and `dtype`

Reply:
Updated